### PR TITLE
Use ApplyDeployment instead of ApplyDeploymentWithForce

### PR DESCRIPTION
**Pull request by knelasevero**
Use ApplyDeployment instead of ApplyDeploymentWithForce 

since in ApplyDeployment it sets an anotation that is the hash of the spec: https://github.com/openshift/library-go/blob/245d1ca16d1514313f3a7ab13ff2c9e3b07b5f43/pkg/operator/resource/resourceapply/apps.go#L95

and ApplyDeployment calls ApplyDeploymentWithForce, which then checks that hash to decide or not to force apply: https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourceapply/apps.go#L131

### CODE REVIEW

File: pkg/generated/clientset/versioned/fake/register.go [hunk 1]
```diff
@@ -21,14 +21,14 @@ var localSchemeBuilder = runtime.SchemeBuilder{
 // AddToScheme adds all types of this clientset into the given scheme. This allows composition
 // of clientsets, like in:
 //
-//   import (
```
Comment (ingvagabund) [inline pkg/generated/clientset/versioned/fake/register.go:24]: Any way to undo this changes?

File: pkg/generated/clientset/versioned/scheme/register.go [hunk 2]
```diff
@@ -21,14 +21,14 @@ var localSchemeBuilder = runtime.SchemeBuilder{
 // AddToScheme adds all types of this clientset into the given scheme. This allows composition
 // of clientsets, like in:
 //
-//   import (
```
[the remark this answers is not in the document]
Comment (knelasevero) [reply pkg/generated/clientset/versioned/scheme/register.go]: Done :heavy_check_mark:

File: pkg/operator/target_config_reconciler.go [hunk 3]
```diff
@@ -117,35 +115,7 @@ func (c TargetConfigReconciler) sync() error {
 		return fmt.Errorf("descheduler should have an interval set")
 	}
 
-	forceDeployment := false
-	_, forceDeployment, err = c.manageConfigMap(descheduler)
```
Comment (ingvagabund) [inline pkg/operator/target_config_reconciler.go:121]: This needs to be handled separately to `ApplyDeployment` as a new annotation. Since the hash calculated from the deployment itself does not include changes in the external dependent resources. There are two cases to take into account:
1. config map changes: the deployment needs to be annotated with `"configmaps/cluster": configMap.ObjectMeta.ResourceVersion` to trigger additional update of the deployment
2. if err != nil, the deployment needs to be scaled down: the original code needs to be kept as it is. Just, without passing `forceDeployment` into `manageDeployment`. `forceDeployment` will be used only for scaling the deployment to `0`.

**Review by ingvagabund — CHANGES_REQUESTED** [review]

File: pkg/operator/target_config_reconciler.go (hunk 3 above)
Comment (knelasevero) [reply pkg/operator/target_config_reconciler.go:121]: I am annotating all the other dependent resources (services, roles, sa, etc). Let me know if we just want to annotate for the configMap

File: pkg/operator/target_config_reconciler.go [hunk 4]
```diff
@@ -146,38 +147,51 @@ func (c TargetConfigReconciler) sync() error {
 					Status: operatorv1.ConditionTrue,
 				}))
 		}
-		return err
```
Comment (ingvagabund) [inline pkg/operator/target_config_reconciler.go:149]: We still want to return `err` here.

File: pkg/operator/target_config_reconciler.go (hunk 3 above)
Comment (ingvagabund) [reply pkg/operator/target_config_reconciler.go:121]: +1 for annotating the other resources

File: pkg/operator/target_config_reconciler.go [hunk 5]
```diff
@@ -146,38 +147,51 @@ func (c TargetConfigReconciler) sync() error {
...
+	} else {
+		specAnnotations["configmaps/cluster"] = configMap.ObjectMeta.ResourceVersion
```
Comment (ingvagabund) [inline pkg/operator/target_config_reconciler.go:151]: Worth double checking https://github.com/openshift/cluster-openshift-controller-manager-operator/commit/5e87d418fa9e170ee615109b9875b92427d7cd16 whether all objects are non-nil. The same for SSO.
